content/dotnet-template-azure-iot-edge-module/CSharp/ModuleBackgroundService.cs

```diff
@@ -0,0 +1,54 @@
+﻿using Microsoft.Azure.Devices.Client;
+using Microsoft.Azure.Devices.Client.Transport.Mqtt;
+using System.Text;
+
+namespace SampleModule;
+
+internal class ModuleBackgroundService : BackgroundService
+{
+    private int counter;
+    private readonly ILogger<ModuleBackgroundService> _logger;
+
+    public ModuleBackgroundService(ILogger<ModuleBackgroundService> logger) => _logger = logger;
+
+    protected override async Task ExecuteAsync(CancellationToken stoppingToken)
+    {
+        MqttTransportSettings mqttSetting = new(TransportType.Mqtt_Tcp_Only);
+        ITransportSettings[] settings = { mqttSetting };
+
+        // Open a connection to the Edge runtime
+        ModuleClient ioTHubModuleClient = await ModuleClient.CreateFromEnvironmentAsync(settings);
+        await ioTHubModuleClient.OpenAsync(stoppingToken);
+        _logger.LogInformation("IoT Hub module client initialized.");
+
+        // Register callback to be called when a message is received by the module
+        await ioTHubModuleClient.SetInputMessageHandlerAsync("input1", PipeMessage, ioTHubModuleClient, stoppingToken);
+    }
+
+    async Task<MessageResponse> PipeMessage(Message message, object userContext)
+    {
+        int counterValue = Interlocked.Increment(ref counter);
+
+        if (userContext is not ModuleClient moduleClient)
+        {
+            throw new InvalidOperationException("UserContext doesn't contain " + "expected values");
```

Comment:
If you keep this approach (passing module client as `userContext`, then I'd fix this string. No reason to concatenate.
```suggestion
            throw new InvalidOperationException("UserContext doesn't contain expected values");
```